ndinterp/src/grid.rs

```diff
@@ -33,9 +48,9 @@ impl Grid {
     /// input at position index as the ratio between the differences dy/dx computed as:
     ///     dy = y_{i} - y_{i-1}
     ///     dx = x_{i} - x_{x-1}
-    pub fn derivative_at(&self, index: usize) -> f64 {
-        let dx = self.input[index] - self.input[index - 1];
-        let dy = self.values[index] - self.values[index - 1];
+    pub fn derivative_at(&'a self, index: usize) -> f64 {
```

Comment:
The annoying part of lifetimes is that you then need to specify them everywhere. Luckily not in calls, but in all types yes...